build-package/certifaiReferenceModelServer/utils/local_server.py

```diff
@@ -7,25 +7,74 @@
 import traceback
 import importlib
 from inspect import getmembers, isfunction
-from flask import Flask, Response, request, jsonify
+from typing import NamedTuple, Dict, List
+
+from flask import Flask, Response, request
 from cortex import Message
+import json
+from werkzeug.exceptions import BadRequest
 
 app = Flask(__name__)
 
 
+class EmptyResponse(NamedTuple):
+    response: Dict = {"predictions": []}
+
+
+class EmptyRequest(NamedTuple):
+    request: Dict = {"payload": {"instances": []}}
+
+
+def is_empty(a):
+    if isinstance(a, List):
+        return all(map(is_empty, a))
+    return False
+
+
+def _validate_message(msg):
+    try:
+        if msg.payload is None:
+            return BadRequest(f'Missing `payload`. Expected format {json.dumps(EmptyRequest().request)}')
+        elif not isinstance(msg.payload, Dict):
+            return BadRequest(
+                f'`payload` must be of type dict. Expected format {json.dumps(EmptyRequest().request)}')
+        elif 'instances' not in msg.payload.keys():
+            return BadRequest(f'Missing `instances`. Expected format {json.dumps(EmptyRequest().request)}')
+        elif not isinstance(msg.payload.get('instances'), List):
+            return BadRequest("`instances` must be of type List[Union[a, List a] ")
```

Comment:
Just a thought, but I think the type annotation reads a little strange because of the generic `a` type. I might suggest just having a sentence instead: "`instances` must be either be an array of instances or a single instance, containing an array of feature values"

Reply:
a is a concrete type (* ) while List a has type * -> * , so in essence instances is List a or(Union) List[List a], but yeah a sentence might be easier for users. I'll change